infra/ballista/build/lambda-handler.py

```diff
@@ -0,0 +1,190 @@
+                process_s.terminate()
+                logging.error(process_s.stdout.read().decode("utf-8"))
+                logging.error(process_s.stderr.read().decode("utf-8"))
+                raise Exception(f"scheduler failed to start after {timeout} seconds")
+    else:
+        logging.error(process_s.stdout.read().decode("utf-8"))
+        raise Exception(
+            f"scheduler failed to init stderror: {process_s.stderr.read().decode('utf-8')}"
+        )
+
+    global process_e
+    process_e = subprocess.Popen(
+        ["/opt/ballista/ballista-executor"],
+        stdout=PIPE,
+        stderr=sys.stderr,
+        bufsize=0,
+    )
+    logging.info("executor starts")
+    # wait till the executor is up and running
+    if process_s.poll() is None:
+        return_code = 1
+        timeout = 0
+        while return_code:
+            res = subprocess.run(
+                ["nc", "-z", "localhost", "50051"], capture_output=True
+            )
+            return_code = res.returncode
+            logging.debug(res)
+            time.sleep(1)
+            timeout += 1
+            if timeout >= 30:
+                process_e.terminate()
+                logging.error(process_e.stdout.read().decode("utf-8"))
+                logging.error(process_e.stderr.read().decode("utf-8"))
+                process_s.terminate()
+                logging.error(process_s.stdout.read().decode("utf-8"))
+                logging.error(process_s.stderr.read().decode("utf-8"))
+                raise Exception(f"executor failed to start after {timeout} seconds")
+    else:
+        logging.error(process_e.stdout.read().decode("utf-8"))
+        logging.error(process_s.stdout.read().decode("utf-8"))
+        raise Exception(
+            f"executor failed to init stderror: {process_e.stderr.read().decode('utf-8')}"
+        )
+
+    global process_cli
+    logging.info("cli starts")
+    process_cli = popen_spawn.PopenSpawn(
+        [
+            "/opt/ballista/ballista-cli",
+            "--host",
+            "localhost",
+            "--port",
+            "50050",
+            "--format",
+            "csv",
+        ]
+    )
+    process_cli.expect(b"\n")
+    logging.debug(process_cli.before)
+
+
+def handler(event, context):
+    """An AWS Lambda handler that runs the provided command with bash and returns the standard output"""
+    global IS_COLD_START
+    is_cold_start = IS_COLD_START
+    IS_COLD_START = False
+
+    start = time.time()
+
+    if is_cold_start:
+        init()
+    init_duration = time.time() - start
+
+    # input parameters
+    logging.debug("event: %s", event)
+    src_command = base64.b64decode(event["cmd"]).decode("utf-8")
+
+    logging.info("command: %s", src_command)
+    if "env" in event:
+        logging.info("env: %s", event["env"])
+        for (k, v) in event["env"].items():
+            os.environ[k] = v
+
+    # process_cli = subprocess.run(command, capture_output=True)
```

Comment:
⁉️ same here as for the comment above

Reply:
Sorry I don't know to which comment you refer

Reply:
very weird! I wrote a comment that doesn't appear anywhere! Maybe I didn't press "send" somehow 😊 sorry for that.

The concern is always the same regarding lines of code commented out without any further explanation: they raise more questions to the reader than they provide information. When I read this, here is what happens in my head:

> what does this line do? why is this commented out? because of performance issues? because it didn't work this way? because it could work this way?

There is no way for me as a reader to answer these questions, so I end up confused.